csi_camera.cpp

```diff
@@ -0,0 +1,451 @@
+                std::shared_ptr<ProtoType> frame_proto = attrs->at("frame_rate");
+                auto frame_value = frame_proto->proto_value();
+                if (frame_value.has_number_value()) {
+                    int frame_num = static_cast<int>(frame_value.number_value());
+                    frame_rate = frame_num;
+                }
+            }
+            if (!frame_rate) {
+                std::cout << "ERROR: frame_rate attribute not found" << std::endl;
+                std::cout << "Setting frame_rate to default value: " << DEFAULT_INPUT_SENSOR<< std::endl;
+                frame_rate = DEFAULT_INPUT_FRAMERATE;
+            }
+            
+            if (attrs->count("video_path") == 1) {
+                std::shared_ptr<ProtoType> video_proto = attrs->at("video_path");
+                auto video_value = video_proto->proto_value();
+                if (video_value.has_string_value()) {
+                    std::string video_str = video_value.string_value();
+                    video_path = video_str;
+                }
+            }
+            if (video_path.empty() ) {
+                std::cout << "ERROR: video_path attribute not found" << std::endl;
+                std::cout << "Setting video_path to default value: " << DEFAULT_INPUT_SENSOR << std::endl;
+                video_path = DEFAULT_INPUT_SENSOR;
+            }
+
+            if (attrs->count("debug") == 1) {
+                std::shared_ptr<ProtoType> debug_proto = attrs->at("debug");
+                auto debug_value = debug_proto->proto_value();
+                if (debug_value.has_bool_value()) {
+                    bool debug_bool = static_cast<bool>(debug_value.bool_value());
+                    debug = debug_bool;
+                }
+            }
+        }
+
+        // OVERRIDE
+
+        void reconfigure(Dependencies deps, ResourceConfig cfg) override {
+            if (debug) {
+                std::cout << "Reconfiguring CSI Camera module" << std::endl;
+            }
+
+            // Update attributes
+            auto attrs = cfg.attributes();
+            validate_attrs(attrs);
+
+            // Stop gst pipeline
+            stop_pipeline();
+
+            // Create GST pipeline
+            std::string pipeline_args = create_pipeline();
+            if (debug) {
+                std::cout << "pipeline_args: " << pipeline_args << std::endl;
+            }
+
+            // Start GST pipeline
+            init_csi(pipeline_args);
+        }
+
+        raw_image get_image(std::string mime_type) override {
+            if (debug) {
+                std::cout << "hit get_image. expecting mime_type " << mime_type << std::endl;
+            }
+            raw_image image;
+            image.mime_type = DEFAULT_OUTPUT_MIMETYPE;
+            if (FAKE_CAMERA) {
+                image.bytes = get_test_image();
+                return image;
+            } else {
+                // TODO: handle no bytes retrieved
```

Comment:
I added a log if no bytes are received.

Not sure what else can be done here since it is possible that an image is simple not available and there is nothing wrong with the pipeline.

The `read` method in RDK camera client will handle the empty frame by erroring during the decode step.

Reply:
Ok cool, sounds like there's nothing to do